### PR TITLE
feat: include namespace in grammar, also concat op

### PR DESCRIPTION
**Pull request by mcmcgrath13**
Update the grammar to have namespaces and the `++` operator

### CODE REVIEW

**Review by OsamaSBCrea — APPROVED** [review]
Nice! thanks

File: src/syntax.grammar [hunk 1]
```diff
@@ -39,6 +39,7 @@ declaration[@isGroup=Declaration]
     | EntityTypeDeclaration
     | ConstraintDeclaration
     | ModuleDeclaration
+    | NamespaceDeclaration
     | WithUseDeclaration
```
Comment (mcmcgrath13) [inline src/syntax.grammar:43]:
```suggestion
    | NamespaceDeclaration
    | WithUseDeclaration
    | ImportDeclaration
```

File: test/Declarations.test.txt [hunk 2]
```diff
@@ -86,3 +102,18 @@ Rel(
...
+# From-Import Declaration
+
+from ns::subns import a, b, c
+
+==>
+
+Rel(
+    ImportDeclaration(
+        Keyword,
+        QualifiedName(BasicId, BasicId),
```
Comment (mcmcgrath13) [inline test/Declarations.test.txt]:
```suggestion
        LhsId(QualifiedName(BasicId, BasicId)),
```

File: test/Declarations.test.txt [hunk 3]
```diff
@@ -86,3 +102,18 @@ Rel(
...
+# From-Import Declaration
+
+from ns::subns import a, b, c
+
+==>
+
+Rel(
+    ImportDeclaration(
+        Keyword,
+        LhsId(QualifiedName(BasicId, BasicId)),
```
Comment (OsamaSBCrea) [inline test/Declarations.test.txt]:
```suggestion
        LhsId(QualifiedName(QualifiedNameId(BasicId), QualifiedNameElem)),
```

**Comment by mcmcgrath13** [comment]
Update the grammar to support namespaces and the ++ operator

**Pull request merged**